geomstats/geometry/connection.py

```diff
@@ -41,39 +49,85 @@ def connection(self, tangent_vector_a, tangent_vector_b, base_point):
         raise NotImplementedError(
                 'connection is not implemented.')
 
-    def exp(self, tangent_vec, base_point):
+    def exp(self, tangent_vec, base_point, n_steps=N_STEPS):
         """Exponential map associated to the affine connection.
 
+        Exponential map at base_point of tangent_vec computed by integration
+        of the geodesic the equation (initial value problem), using the
+        christoffel symbols
+
         Parameters
         ----------
         tangent_vec: array-like, shape=[n_samples, dimension]
                                  or shape=[1, dimension]
 
         base_point: array-like, shape=[n_samples, dimension]
                                 or shape=[1, dimension]
-        """
-        raise NotImplementedError(
-                'The affine connection exponential is not implemented.')
 
-    def log(self, point, base_point):
+        n_steps: int
+        """
+        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+
+        def geodesic_equation(x, v):
```

Comment:
Avoid single letters to denote variables, otherwise it becomes impossible to use automatic find and replace later

Reply:
Why did you choose to have geodesic_equation as an inner function, and not a method of the class connection?

Reply:
It wasn't used anywhere else. In fact it outputs the acceleration, it can be renamed if keeping geodesic_equation as a method is important, but I don't think it was useful: how do you output an ODE?

Reply:
It can be useful during inheritance, if we want to have a geodesic_equation method for LeviCivita connection (if we keep this class), or RiemannianMetric (if it inherits from Connection, like in the code refactorization that we have discussed with @opeltre )

Reply:
yep we'll definitely have to see how to put some order in all of this! 
In the meantime it's good to have it working at some place so that we only have to re-reference or copy-paste when it's figured out :)

We worked on some other kind of cool ODE methods with nguigs yesterday using autograd! it may be good to have all these in a separate file and import them as needed, so that we don't mix up autograd-related ODE integration techniques with the code defining the Connection class interface...

Reply:
Agreed! @AliceLeBrigant and I were also talking about creating an integration_utils.py somewhere.

Reply:
Agreed! @AliceLeBrigant and I were also talking about creating an integration_utils.py somewhere.